### PR TITLE
Prohibit NaN and infinite values in strict JSON mode

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: runtime/common/src/main/kotlin/kotlinx/serialization/json/JSON.kt [hunk 1]
```diff
@@ -17,15 +17,15 @@
...
-        private val unquoted: Boolean = false,
-        private val indented: Boolean = false,
-        private val indent: String = "    ",
-        internal val nonstrict: Boolean = false,
-        val updateMode: UpdateMode = UpdateMode.OVERWRITE,
-        val context: SerialContext? = null
+    private val unquoted: Boolean = false,
+    private val indented: Boolean = false,
+    private val indent: String = "    ",
+    internal val strictMode: Boolean = true,
```
Comment (sandwwraith) [inline runtime/common/src/main/kotlin/kotlinx/serialization/json/JSON.kt]: Can we afford such breaking change in API?

Comment (sandwwraith) [reply runtime/common/src/main/kotlin/kotlinx/serialization/json/JSON.kt]: There will be at least one supporting release (together with 1.2.60), so seems this should be postponed until core refactoring?

Comment (qwwdfsad) [reply runtime/common/src/main/kotlin/kotlinx/serialization/json/JSON.kt]: Yes, I'm fine with it

**Comment by sandwwraith** [comment]
Corresponding [docs](https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/runtime_usage.md#json) should be updated, too

**Comment by sandwwraith** [comment]
Merged manually: https://github.com/Kotlin/kotlinx.serialization/commit/63d223960ff6f80fdb030f484910d45e7f6524e4

**Pull request closed without merging**